calm-integrations/ey-calm-migration-scripts/ey-recreate-categories.py

```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# TODO- Add authentication
+# TODO- Add skip registration flag
+
+# pylint: disable= E402
+
+import gevent.monkey
+gevent.monkey.patch_all()
+import sys
+import logging
+import warnings
+import os
+import requests
+import json
+from IPython.frontend.terminal.embed import InteractiveShellEmbed
+
+from calm.common.config import init_config, get_config_dict, get_config
+cfg = init_config()
+
+from calm.common.flags import gflags
+gflags.FLAGS(sys.argv)
+
+from calm.lib.model.store.db import create_db_connection
+from calm.pkg.common.scramble import init_scramble
+from calm.lib.model.store.db_session import create_session, flush_session, set_session_type
+import calm.lib.model as model
+
+warnings.filterwarnings("ignore")
+log = logging.getLogger('cshell')
+logging.basicConfig(level=logging.INFO,
+                    format="%(message)s",
+                    datefmt='%H:%M:%S')
+
+SOURCE_PC_IP = "10.44.19.70"
+DESTINATION_PC_IP = "10.46.7.50"
+PC_PORT = 9440
+LENGTH = 100
+
+source_base_url = "https://{}:{}/api/nutanix/v3".format(SOURCE_PC_IP,str(PC_PORT))
+dest_base_url = "https://{}:{}/api/nutanix/v3".format(DESTINATION_PC_IP,str(PC_PORT))
+source_pc_auth = { "username": os.environ['SOURCE_PC_USER'], "password": os.environ['SOURCE_PC_PASS']}
+dest_pc_auth = { "username": os.environ['DEST_PC_USER'], "password": os.environ['DEST_PC_PASS']}
+
+user_categories_list = []
+
+CATEGORY_KEY_LIST = [
+    "CalmApplication",
+    "CalmDeployment",
+    "CalmService",
+    "CalmPackage",
+    "OSType",
+    "CalmVmUniqueIdentifier",
+    "CalmUsername",
+    "account_uuid"
+]
+
+headers = {'content-type': 'application/json', 'Accept': 'application/json'}
+
+def init_contexts():
+    cfg = get_config()
+    keyfile = cfg.get('security', 'keyfile')
+    init_scramble(keyfile)
+    set_session_type('green', cfg.get('store', 'flush_parallelisation_factor'), cfg.get('store', 'bulk_size'))
+    create_db_connection(register_entities=False)
+
+def create_category(base_url, auth, key):
+    method = 'PUT'
+    url = base_url + "/categories/{}".format(key)
+    payload = {
+        "name": key
+    }
+    resp = requests.request(
+            method,
+            url,
+            data=json.dumps(payload),
+            headers=headers,
+            auth=(auth["username"], auth["password"]),
+            verify=False
+    )
+    if resp.ok:
+        return True
+    else:
+        print("Failed to create category key '{}'.".format(key))
+        print('Status code: {}'.format(resp.status_code))
+        print('Response: {}'.format(json.dumps(json.loads(resp.content), indent=4)))
+        exit(1)
```

Comment:
Better to raise exception instead of exit(1).

Reply:
done